torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3136,6 +3136,20 @@ struct to_ir {
         return emitBuiltinCall(
             tree->range(), *method.graph(), kind, named_values, {});
       }
+      case '%': {
+        auto lhs = emitSugaredExpr(Expr(tree->tree(0)), 0)
+                       ->asValue(tree->tree(0)->range(), method);
+        auto lhs_type = lhs->type();
+        if (lhs_type == StringType::get()) {
+          auto values = getValues(tree->trees(), /*maybe_unpack=*/false);
+          auto node = graph->create(aten::percentFormat, values, 1)
+                          ->setSourceRange(tree->range());
+          Value* output = graph->insertNode(node)->output();
+          output->setType(StringType::get());
+          return output;
+        }
+        // else, fall through to the next case
```

Comment:
Thinking more about it, I am really nervous about this fallthrough case since it can be easily broken. Could you refactor code that handles the next case into a function, call that function here and don't fallthrough?